db/migrate/20161129111100_add_wopi.rb

```diff
@@ -0,0 +1,53 @@
+class AddWopi < ActiveRecord::Migration
+  def up
+
+    add_column :assets, :lock, :string, limit: 1024
+    add_column :assets, :lock_ttl, :integer
+    add_column :assets, :version, :integer, default: 1
+
+    create_table :wopi_discoveries do |t|
+      t.integer :expires, null: false
+      t.string :proof_key_mod, null: false
+      t.string :proof_key_exp, null: false
+      t.string :proof_key_old_mod, null: false
+      t.string :proof_key_old_exp, null: false
+    end
+
+    create_table :wopi_apps do |t|
+      t.string :name, null: false
+      t.string :icon, null: false
+      t.integer :wopi_discovery_id, null: false
+    end
+
+    create_table :wopi_actions do |t|
+      t.string :action, null: false
+      t.string :extension, null: false
+      t.string :urlsrc, null: false
+      t.integer :wopi_app_id, null: false
+    end
+
+    create_table :tokens do |t|
+      t.string :token, null: false
+      t.integer :ttl, null: false
+      t.integer :user_id, null: false
```

Comment:
t.references :user will do the job here and will generate a foreign key or even better t.belongs_to(:user, foreign_key: true)